### PR TITLE
feat: dynamic search support for dropdown widget in smart action forms

### PR DESCRIPTION
**Pull request by realSpok**
## Definition of Done
#85yxm8ndn
### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

File: packages/datasource-customizer/src/decorators/actions/context/base.ts [hunk 1]
```diff
@@ -31,11 +32,13 @@ export default class ActionContext<
     filter: TFilter<S, N>,
     used?: Set<string>,
     changedField?: string,
+    searchValue?: string,
   ) {
     super(collection, caller);
     this.formValues = formValue;
     this.filter = filter;
     this.changedField = changedField;
+    this.searchValue = searchValue;
```
Comment (ghusse) [inline packages/datasource-customizer/src/decorators/actions/context/base.ts:41]: 🤔 is there a way not to add this property to the every contexts, but only the context used in `options` for `dropdown`?

Comment (realSpok) [reply packages/datasource-customizer/src/decorators/actions/context/base.ts:57]: I have been trying some changes in the types definition to make it accept a different context type in the `options` handler, but to no avail. 
Another option that we could consider is to name the search handler differently than options. This allows to specify a different type. ex: `searchHandler` or `optionsHandler`

**Comment by realSpok** [comment]
As discussed, I have updated the search handler to take 2 separate parameters: `context` and `searchValue`

File: packages/datasource-customizer/src/decorators/actions/collection.ts [hunk 2]
```diff
@@ -163,7 +168,11 @@ export default class ActionCollectionDecorator extends CollectionDecorator {
...
   }
 
-  private async evaluate<T>(context: ActionContext, value: ValueOrHandler): Promise<T> {
-    return typeof value === 'function' ? value(context) : value;
+  private async evaluate<T>(
+    context: ActionContext,
+    searchValue: string | null,
+    value: ValueOrHandler,
+  ): Promise<T> {
+    return typeof value === 'function' ? value(context, searchValue) : value;
```
[the remark this answers is not in the document]
Comment (realSpok) [reply packages/datasource-customizer/src/decorators/actions/collection.ts]: I have made the change

**Review by ghusse — APPROVED** [review]
- ✅&nbsp;Code review
  - ✅&nbsp; Use pure functions when possible
  - ✅&nbsp; Performance concerns
  - ✅&nbsp; Security concerns
  - ✅&nbsp; The PR tackle one subject only
- ✅&nbsp;Automatic tests
  - ✅&nbsp; Unit tests
  - ✖️&nbsp; Integration tests
- ✅&nbsp;Manual tests
- ✅&nbsp;PR title
- ✅&nbsp;Destination branch
- ✅&nbsp;PR linked to the clickup task

**Pull request merged**